optimizely/helpers/validator.py

```diff
@@ -189,3 +191,57 @@ def is_attribute_valid(attribute_key, attribute_value):
     return True
 
   return False
+
+
+def is_finite_number(value):
+  """ Method to validate if the given value is a number and not one of NAN, INF, -INF
+
+  Args:
+    value: Value to be validated
+
+  Returns:
+    Boolean: True if value is a number and not NAN, INF or -INF else False
+  """
+  if not isinstance(value, (int, float)):
+      # numbers.Integral instead of int to accomodate long integer in python 2
+    return False
+
+  if isinstance(value, bool):
+    # bool is a subclass of int
+    return False
+
+  if isinstance(value, float):
+    if math.isnan(value) or math.isinf(value):
+      return False
+
+  return True
```

Comment:
Thanks for extracting this! As mentioned on a now-obsolete commit, can we

1. enforce an absolute size limit of 1e53 if `isinstance(value, numbers.Integral)`
2. update `helpers.validator.is_attribute_valid` such that it calls `helpers.validator.is_finite_number`

?

Reply:
And @aliabbasrizvi's addendum:

> Random. I am wondering if we should extend this validation i.e. the `is_finite` check to revenue and quantity values?

+1, is worth considering

Reply:
Will do this in a separate commit.

Reply:
I assume by "separate commit" you mean "separate PR"? That's fine, thanks! Tracking in 3654